### ISSUE TITLE
POS will be all "ADP" when the inputs have empty string when using .pipe()

### ISSUE THREAD

**Issue by xDarkLemon** [explosion/spaCy#7029]
## How to reproduce the behaviour
When I use `spacy_model.pipe()` to process documents in parallel, the POS tagging will give wrong result if there is any empty string int the documents.
For example:
(1) Normally:
```
docs=["Hello world!","How are you?"]
spacy_docs=spacy_model.pipe(docs)
for spacy_doc in spacy_docs:
    for sentence in spacy_doc.sents:
        print([(token.text,token.pos_) for token in sentence])
```
Output:
```
[('Hello', 'INTJ'), ('world', 'NOUN'), ('!', 'PUNCT')]
[('How', 'ADV'), ('are', 'VERB'), ('you', 'PRON'), ('?', 'PUNCT')]
```
(2) Input has empty string:
```
docs=["Hello world!","How are you?",""]
spacy_docs=spacy_model.pipe(docs)
for spacy_doc in spacy_docs:
    for sentence in spacy_doc.sents:
        print([(token.text,token.pos_) for token in sentence])
```
Output:
```
[('Hello', 'ADP'), ('world', 'ADP'), ('!', 'ADP')]
[('How', 'ADP'), ('are', 'ADP'), ('you', 'ADP'), ('?', 'ADP')]
```
In my opinion, the empty string should not affect the output of other documents. I think there is bug.

### Info about spaCy
- **spaCy version:** 3.0.1
- **Platform:** Darwin-18.7.0-x86_64-i386-64bit
- **Python version:** 3.6.10
- **Pipelines:** en_core_web_sm (3.0.0)

* Operating System: MacOS and Linux
* Python Version Used: python 3.6 and python 3.7
* spaCy Version Used: 3.0.1
* Environment Information:

**Comment by adrianeboyd** [comment]
Thanks for the report! I can reproduce this and it definitely looks like a bug. We will look into it!

**Comment by PradipNichite** [comment]
Hi @adrianeboyd , anyone working on this? I would like to contribute.

**Comment by honnibal** [comment]
Hi @PradipNichite ,

If you want to have a look at this, please do go ahead! It's possible Adriane is looking into it as well but it's almost the end of the work day in Germany. In any case, if you develop tests for the bug, they're unlikely to be the same as the ones someone else develops, so the effort probably won't be wasted.

I think the bug will almost certainly be within the `spacy-transformers` package, likely to do with the alignment. You might be able to see something wrong if you look at the `doc._.trf_data.wordpieces` object, especially the `doc._.trf_data.wordpieces.align`. Here are some source links that might be helpful:

* The `WordpieceBatch` data structure: https://github.com/explosion/spacy-transformers/blob/master/spacy_transformers/data_classes.py#L16
* The `TransformerData` data structure: https://github.com/explosion/spacy-transformers/blob/master/spacy_transformers/data_classes.py#L131
* The function where the interaction with the transformer happens: https://github.com/explosion/spacy-transformers/blob/master/spacy_transformers/layers/transformer_model.py#L116
* The place where the transformer data is set onto the `Doc` object: https://github.com/explosion/spacy-transformers/blob/master/spacy_transformers/pipeline_component.py#L224

**Comment by PradipNichite** [comment]
Hi @honnibal , Thanks for the help. I am new to open source contribution and first time with Spacy.  I might take more time.
I will try to understand the bug. if @adrianeboyd  already looking into it then fine. I will look for another issue, please let me know if you find any Good first Issue.

**Comment by adrianeboyd** [comment]
No, this is really bad and happens with all models. The other docs in the same batch as the empty doc have messed up predictions across all components:

```python
import spacy
nlp = spacy.load("en_core_web_sm")

texts = ["a", "a", "a", "a", "a", "a", "a", ""]
docs1 = list(nlp.pipe(texts, batch_size=1))
docs2 = list(nlp.pipe(texts, batch_size=4))
print([doc[0].tag_ for doc in docs1[:-1]])
print([doc[0].tag_ for doc in docs2[:-1]])
```

```none
['DT', 'DT', 'DT', 'DT', 'DT', 'DT', 'DT']
['DT', 'DT', 'DT', 'DT', 'IN', 'IN', 'IN']
```

I haven't figured it out yet, but I'm slow at debugging thinc internals.

**Comment by adrianeboyd** [comment]
It's something in the tok2vec, so actually not at all with the transformers models.

**Comment by adrianeboyd** [comment]
The input to the softmax layer is 0s for the whole batch.
